data/frameProcessor/src/ExcaliburProcessPlugin.cpp

```diff
@@ -25,11 +25,13 @@ namespace FrameProcessor
       image_width_(2048),
       image_height_(256),
       image_pixels_(image_width_ * image_height_),
-      packets_lost_(0)
+      packets_lost_(0),
+      number_of_fems_(6)
```

Comment:
I wonder if we should have these initial parameters (# FEMs, width, height) set as named constants that refer to it being a 3M detector rather than magic numbers?

Reply:
Good point.

Also, is `image_height_` supposed to be for a FEM, or the whole detector? If the latter the defaults aren't actually valid (height 256, fems 6).

Reply:
I think these values (width and height) are overwritten by configurations aren't they?  Or else I don't see how this would work for 1M or 3M

Reply:
Yes those values for width and height are defaults for a minimal system, i.e. a single FEM, and are always overwritten by the configuration. We should make them self-consistent and refer to the same scale of system. I'd suggest since a single FEM is the 'quantum' of readout, stick with that as the default for `number_of_fems_` which I think then gets modified by the header of incoming frames, rather than being updated from configuration. That variable only affects the packet loss counting I think rather than the scale of the image?

Reply:
I have just updated it using constants from `ExcaliburProcessPlugin.h`, but then I thought should these numbers include gaps?

Reply:
No, the gaps shouldn't be included here as the reordering algorithm in this plugin doesn't insert them